bin/install_clamav_on_mac

```diff
@@ -0,0 +1,148 @@
+sudo mkdir -p "$TMP_FOLDER"
+sudo mkdir -p "$RUN_FOLDER"
+
+[ -e "$CLAMD_LOG_FILE" ] || sudo touch "$CLAMD_LOG_FILE"
+[ -e "$CLAMD_ERROR_LOG_FILE" ] || sudo touch "$CLAMD_ERROR_LOG_FILE"
+[ -e "$FRESHCLAM_LOG_FILE" ] || sudo touch "$FRESHCLAM_LOG_FILE"
+[ -e "$FRESHCLAM_ERROR_LOG_FILE" ] || sudo touch "$FRESHCLAM_ERROR_LOG_FILE"
+[ -e "$CLAMD_SOCKET_FILE" ] || sudo touch "$CLAMD_SOCKET_FILE"
+
+sudo chown -R root:wheel "$CONFIG_FOLDER"
+sudo chown -R clamav:clamav "$DB_FOLDER"
+sudo chown -R clamav:clamav "$TMP_FOLDER"
+sudo chown -R clamav:clamav "$RUN_FOLDER"
+sudo chown clamav:clamav "$CLAMD_LOG_FILE" "$CLAMD_ERROR_LOG_FILE" "$FRESHCLAM_LOG_FILE" "$FRESHCLAM_ERROR_LOG_FILE"
+sudo chmod 0644 "$CLAMD_CONFIG_FILE" "$FRESHCLAM_CONFIG_FILE"
+sudo chmod 0644 "$CLAMD_LOG_FILE" "$CLAMD_ERROR_LOG_FILE" "$FRESHCLAM_LOG_FILE" "$FRESHCLAM_ERROR_LOG_FILE"
+
+DAEMON_FOLDER=/Library/LaunchDaemons
+CLAMD_DAEMON_NAME=clamav.clamd
+CLAMD_DAEMON_FILE=$DAEMON_FOLDER/$CLAMD_DAEMON_NAME.plist
+FRESHCLAM_DAEMON_NAME=clamav.freshclam
+FRESHCLAM_DAEMON_FILE=$DAEMON_FOLDER/$FRESHCLAM_DAEMON_NAME.plist
+
+sudo tee "$CLAMD_DAEMON_FILE" << EOF > /dev/null
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>Label</key>
+    <string>${CLAMD_DAEMON_NAME}</string>
+    <key>ProgramArguments</key>
+    <array>
+        <string>$(brew --prefix)/sbin/clamd</string>
+        <string>--foreground</string>
+    </array>
+    <key>KeepAlive</key>
+    <true/>
+    <key>StandardErrorPath</key>
+    <string>${CLAMD_ERROR_LOG_FILE}</string>
+</dict>
+</plist>
+EOF
+
+sudo tee "$FRESHCLAM_DAEMON_FILE" << EOF > /dev/null
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>Label</key>
+    <string>${FRESHCLAM_DAEMON_NAME}</string>
+    <key>ProgramArguments</key>
+    <array>
+        <string>$(brew --prefix)/bin/freshclam</string>
+        <string>--daemon</string>
+        <string>--foreground</string>
+    </array>
+    <key>KeepAlive</key>
+    <true/>
+    <key>StandardErrorPath</key>
+    <string>${FRESHCLAM_ERROR_LOG_FILE}</string>
+</dict>
+</plist>
+EOF
+
+sudo chown root:wheel "$CLAMD_DAEMON_FILE" "$FRESHCLAM_DAEMON_FILE"
+sudo chmod 0644 "$CLAMD_DAEMON_FILE" "$FRESHCLAM_DAEMON_FILE"
+
+# restart (changes may have been made)
+sudo launchctl unload "$CLAMD_DAEMON_FILE" "$FRESHCLAM_DAEMON_FILE"
+sudo launchctl load "$CLAMD_DAEMON_FILE" "$FRESHCLAM_DAEMON_FILE"
+
+# wait til port being listened too
+start=$(date +%s)
+printf "Restarting clamav, waiting for clamd to listen on port $CLAMD_TCP_SOCKET"
+while ! sudo lsof -PiTCP -sTCP:LISTEN | grep ".*clamd.*$CLAMD_TCP_SOCKET" > /dev/null; do
+  printf "."
+
+  if (( $(date +%s)-start > 60 ))
+  then
+    printf "timed out!\n"
+    break;
+  fi
+
+  sleep 5;
+done;
```

Comment:
WTF...

Reply:
something we used in apply and i had tweaked to, generally, just work. Have you tried it?

Reply:
this bit is just waiting until clamav restarts with the new config.